internal/providers/aws/aws.go

```diff
@@ -68,3 +61,13 @@ func NewFetcher(l *log.Logger) (resource.Fetcher, error) {
 		AWSSession: sess,
 	}, nil
 }
+
+func Init(f *resource.Fetcher) error {
+	// Determine the partition and region this instance is in
+	regionHint, err := ec2metadata.New(f.AWSSession).Region()
+	if err != nil {
+		regionHint = "us-east-1"
+	}
+	f.S3RegionHint = regionHint
```

Comment:
Perhaps this should actually be serialized into e.g. `/run/ignition-ec2.json` (much like the fetch stage does) so that it's automatically shared by later invocations?

Reply:
Possibly.  Unlike with config fetch, the query should be relatively inexpensive and there's no risk that the answer will change between calls.  But it might be a worthwhile optimization.